arbnode/forwarder.go

```diff
@@ -186,3 +205,162 @@ func (f *TxDropper) StopAndWait() {}
 func (f *TxDropper) Started() bool {
 	return true
 }
+
+type RedisTxForwarder struct {
+	stopwaiter.StopWaiterSafe
+
+	config *ForwarderConfig
+	errors int
+
+	mtx              sync.RWMutex
+	forwarder        *TxForwarder
+	redisCoordinator *redisutil.RedisCoordinator
+	currentTarget    string
+	fallbackTarget   string
+}
+
+func NewRedisTxForwarder(fallbackTarget string, config *ForwarderConfig) *RedisTxForwarder {
+	return &RedisTxForwarder{
+		config:         config,
+		fallbackTarget: fallbackTarget,
+	}
+}
+
+func (f *RedisTxForwarder) PublishTransaction(ctx context.Context, tx *types.Transaction) error {
+	forwarder := f.getForwarder()
+	if forwarder == nil {
+		return ErrNoSequencer
+	}
+	return forwarder.PublishTransaction(ctx, tx)
+}
+
+func (f *RedisTxForwarder) CheckHealth(ctx context.Context) error {
+	forwarder := f.getForwarder()
+	if forwarder == nil {
+		return ErrNoSequencer
+	}
+	return forwarder.CheckHealth(ctx)
+}
+
+// not thread safe vs update and itself
+func (f *RedisTxForwarder) Initialize(ctx context.Context) error {
+	var err error
+	f.redisCoordinator, err = redisutil.NewRedisCoordinator(f.config.RedisUrl)
+	if err != nil {
+		return errors.Wrap(err, "unable to create redis coordinator")
+	}
+	return nil
```

Comment:
Let's call `update` here to make sure that before the node starts up it's given at least one chance to get an initial forwarding target.